torch/csrc/jit/mobile/debug_info.cpp

```diff
@@ -0,0 +1,52 @@
+#include <torch/csrc/jit/mobile/debug_info.h>
+#include <torch/csrc/jit/serialization/source_range_serialization.h>
+
+#include <ATen/core/ivalue.h>
+#include <torch/csrc/jit/serialization/pickle.h>
+
+#include <c10/util/string_view.h>
+
+namespace torch {
+namespace jit {
+
+MobileDebugTable::MobileDebugTable(
+    std::unique_ptr<caffe2::serialize::PyTorchStreamReader>& reader) {
+  const std::vector<std::string>& record_names = reader->getAllRecords();
+  const c10::string_view suffix("debug_pkl");
+  for (const auto& record_name : record_names) {
+    if (c10::string_view(record_name.data(), record_name.size())
+            .ends_with(suffix)) {
+      at::DataPtr debug_data;
+      size_t debug_size{0};
+      std::tie(debug_data, debug_size) = reader->getRecord(record_name);
+      auto ivalues =
+          jit::unpickle(
+              reinterpret_cast<const char*>(debug_data.get()), debug_size)
+              .toTuple()
+              ->elements();
+      std::unique_ptr<SourceRangeDeserializer> deserializer =
+          std::make_unique<SourceRangeDeserializer>();
+      for (auto& val : ivalues) {
+        auto tup_elems = val.toTuple()->elements();
+        TORCH_CHECK(
+            tup_elems.size() == 3,
+            "Source debug tuple must have three elements:"
+            "byte_offset, source_tag, source_range");
+        int64_t debug_handle = tup_elems[1].toInt();
+        auto source_range = deserializer->deserialize(tup_elems[2]);
+        source_range_map_.emplace(debug_handle, std::move(source_range));
+      }
+    }
```

Comment:
Right now in torchscript, usually, each class has its own code.py file and each such file has corresponding debug_pkl. So we have load each of them. We can later consolidate all that. BTW I think you asked somewhere, if we can strip off debug info and I think that is possible.

Reply:
So consolidation can be done only for on device usecases. For TorchScript models we cannot because debug_pkl stores Source information as a tuple of (byte_offset, source range). byte_offset here is the offset in the code.py file, so to ensure one-to-one mapping every code file needs its own debug_pkl. For mobile we are generating model wide unique source range tag so we dont need to have separate debug_pkl files.